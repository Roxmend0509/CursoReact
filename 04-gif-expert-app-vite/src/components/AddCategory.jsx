import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChage = (event) =>{
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const trimInputValue= inputValue.trim();
        if(trimInputValue.length <= 1) return;
        //onAddCategory((categories) => [ inputValue, ...categories ])
        setInputValue('');
        onNewCategory(trimInputValue)
    }

  return (
    <form onSubmit={ onSubmit }>
    <input 
        type="text"
        placeholder="Buscar GIF's"
        value={ inputValue }
        onChange={ (event) => onInputChage(event) }
    />
    </form>
  )
}
