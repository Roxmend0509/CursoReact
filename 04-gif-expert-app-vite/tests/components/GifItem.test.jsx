import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe(' Pruebas en <GifItem />',() => {

    test('Debe hacer match con el snapshot', () => { 
        const title = 'Hardy'
        const url = 'https://media4.giphy.com/media/KDQ25pahVfwGRvvT9X/giphy.gif?cid=77f5f3b1aa2bpja40g7trcb4hnxhhp33h0tx4jy0mufyaulp&rid=giphy.gif&ct=g'
        
        const {container} = render(<GifItem title={title} url={url} />)

        expect(container).toMatchSnapshot();
     })

})