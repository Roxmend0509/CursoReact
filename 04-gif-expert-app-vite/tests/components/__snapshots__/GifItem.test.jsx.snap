// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hardy"
      src="https://media4.giphy.com/media/KDQ25pahVfwGRvvT9X/giphy.gif?cid=77f5f3b1aa2bpja40g7trcb4hnxhhp33h0tx4jy0mufyaulp&rid=giphy.gif&ct=g"
    />
    <p>
      Hardy
    </p>
  </div>
</div>
`;
